sympy/stats/stochastic_process_types.py

```diff
@@ -638,11 +704,33 @@ def transition_probabilities(self):
         """
         return self.args[2]
 
+    @property
+    def num_states(self):
+        """
+        The number of states in the Markov Chain. Can be symbolic.
+        """
+        return self.transition_probabilities.shape[0] \
+            if self.transition_probabilities is not None else None
```

Comment:
I think that a normal if/else is clearer than a multiline ternary conditional expression